Tasks/AzurePowerShell/Utility.ps1

```diff
@@ -0,0 +1,46 @@
+function Update-PSModulePathForHostedAgent {
+    [CmdletBinding()]
+    param([string] $targetAzurePs)
+
+    if ($targetAzurePs) {
+        $hostedAgentAzureRmModulePath = $env:SystemDrive + "\Modules\AzureRm_" + $targetAzurePs
+        $hostedAgentAzureModulePath = $env:SystemDrive + "\Modules\Azure_" + $targetAzurePs
+    }
+    else {
+        $hostedAgentAzureRmModulePath = Get-LatestModule -patternToMatch "^azurerm_[0-9]+\.[0-9]+\.[0-9]+$" -patternToExtract "[0-9]+\.[0-9]+\.[0-9]+$"
+        $hostedAgentAzureModulePath  =  Get-LatestModule -patternToMatch "^azure_[0-9]+\.[0-9]+\.[0-9]+$"   -patternToExtract "[0-9]+\.[0-9]+\.[0-9]+$"
+    }
+    $env:PSModulePath = $hostedAgentAzureRmModulePath + ";" + $env:PSModulePath
```

Comment:
adding logs will be better here like updating the module path, updated module path